docs/src/plugins/sidebar-scroll-into-view/index.js

```diff
@@ -0,0 +1,51 @@
+// Fixes an issue where the sidebar is not scrolled into view on navigation.
+//
+//     https://github.com/facebook/docusaurus/issues/823
+//
+// USAGE
+//
+//     Include the plugin in your `docusaurus.config.js`:
+//
+//          plugins: [
+//            require.resolve('./src/plugins/sidebar-scroll-into-view'),
+//          ],
+//
+
+const INNER_HTML = `
+  function handleScrollIntoView() {
+    const observer = new MutationObserver(() => {
+    const element = document.querySelector('aside .menu__link--active');
+      if (element) {
+        observer.disconnect();
+        element.scrollIntoView();
+      }
+    });
+
+    observer.observe(document.body, { childList: true, subtree: true });
+  }
+
+  document.addEventListener("DOMContentLoaded", function() {
+    handleScrollIntoView();
+  });
+  window.addEventListener('popstate', () => handleScrollIntoView());
```

Comment:
https://github.com/dagster-io/internal/blob/e419fc8aac53fe488d35ce480d8f20a70831e4fc/dagster-cloud/js_modules/app-cloud/src/performance/performance.tsx#L163-L172

Reply:
You need to override pushState because there is no real pushstate event

Reply:
https://caniuse.com/mdn-api_navigation 
navigation API would be the way to go but not supported in safari/firefox

Reply:
ah, that's good to know

Reply:
Wait I lied, I didn't comment all the code that I thought I did.